helix-view/src/document.rs

```diff
@@ -933,7 +933,10 @@ impl Document {
                     "Path is read only"
                 ));
             }
-            let backup = if path.exists() {
+
+            // Effectively ignores hardlink check if the metadata cant be read (e.g. on certain Windows configurations)
+            let is_hardlink = helix_stdx::get_hardlink_count(&write_path).unwrap_or(1) > 1;
+            let backup = if path.exists() && !is_hardlink {
```

Comment:
We should still be making a backup for handlinks (to prevent file loss) but it should be done by copying the file instead of moving (equivalent of bufcopy)